js/signin.js

```diff
@@ -0,0 +1,104 @@
+const emailInput = document.querySelector("#email-input");
+const passwordInput = document.querySelector("#password-input");
+const emailErrorMessage = document.querySelector("#email-error-message");
+const passwordErrorMessage = document.querySelector("#password-error-message");
+const loginForm = document.querySelector("#login-form");
+
+const eyeOffButton = document.querySelector(".eye-off");
+const eyeOnButton = document.querySelector(".eye-on");
+
+const EMAIL_REGEX = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,6}$/;
+
+// 이메일 입력
+emailInput.addEventListener("focusout", checkEmailInput);
+emailInput.addEventListener("focus", resetInputStyle);
+
+function checkEmailInput() {
+  if (emailInput.value.trim() === "") {
+    showEmailErrorMessage("이메일을 입력해 주세요.");
+  } else if (!validateEmail(emailInput.value.trim())) {
+    showEmailErrorMessage("올바른 이메일 주소가 아닙니다.");
+  } else {
+    hideErrorMessage();
+  }
+}
+
+function validateEmail(email) {
+  return EMAIL_REGEX.test(email);
+}
+
+function showEmailErrorMessage(message) {
+  emailErrorMessage.textContent = message;
+  emailErrorMessage.style.display = "block";
+  emailInput.classList.add("sign-input-error");
+}
+
+function hideErrorMessage() {
+  emailErrorMessage.style.display = "none";
+  emailInput.classList.remove("sign-input-error");
+}
+
+function resetInputStyle() {
+  hideErrorMessage();
+}
+
+// 비밀번호 입력
+passwordInput.addEventListener("focusout", checkPasswordInput);
+passwordInput.addEventListener("focus", resetPasswordInputStyle);
+
+function checkPasswordInput() {
+  if (passwordInput.value.trim() === "") {
+    showPasswordErrorMessage("비밀번호를 입력해 주세요.");
+  } else {
+    hidePasswordErrorMessage();
+  }
+}
```

Comment:
요런 느낌도 괜찮을 것 같아요! 
위에서는 중복으로 사용되기 때문에 별도의 변수로 선언한다고 말씀드렸는데 꼭 중복으로 사용되지 않더라도 의미를 조금 더 명확하게 드러내기 위해서 변수로 선언해주셔도 괜찮습니다!

```suggestion
function checkPasswordInput() {
  const password = passwordInput.value.trim();
  
  if (password === "") {
    showPasswordErrorMessage("비밀번호를 입력해 주세요.");
    return;
  }
  
  hidePasswordErrorMessage();
}
```

Reply:
email쪽에 등록하는 이벤트 핸들러 함수와도 같은 맥락인데 생각해보니 `checkPasswordInput`은 현재 유효성을 검증한다는 의미를 담기에는 조금 아쉬운 것 같기도 한데요~ `validatePasswordInput`같은 느낌도 괜찮을 것 같아요!
```suggestion
function validatePasswordInput() {
  if (passwordInput.value.trim() === "") {
    showPasswordErrorMessage("비밀번호를 입력해 주세요.");
  } else {
    hidePasswordErrorMessage();
  }
}
```